subsys/bluetooth/host/Kconfig

```diff
@@ -398,12 +398,24 @@ config BT_AUTO_PHY_UPDATE
 	depends on BT_PHY_UPDATE
 	default y if !BT_USER_PHY_UPDATE
 	help
-	  Initiate PHY Update Procedure on connection establishment.
+	  Initiate PHY Update Procedure on connection establishment. This will attempt
+	  to update the connection to use 2M PHY, however it doesn't actually guarantee
+	  that this is what will be used in the end.
```

Comment:
We should take future PHYs into consideration. 

One possible solution would be to deprecate this, and introduce 2 new Kconfig options similar to the one you have below: One for central, and one for peripheral, and the make them `choice`s where the choices are something like 

```
None
1M
2M
Coded
```

where `None` would disable the auto PHY update. This will allow users to choose the PHY preference we send to the controller, as well as being expandable with future PHYs. 

Of course, at this point, it may just be easier to refer to the `bt_conn_le_phy_update` API function, and deprecate `BT_AUTO_PHY_UPDATE`

Reply:
> Of course, at this point, it may just be easier to refer to the bt_conn_le_phy_update API function, and deprecate BT_AUTO_PHY_UPDATE

No objections from me, but I got the feeling at least @nordicjm would oppose this opinion :) (which some reasonable justifications)